src/azureHelpers.ts

```diff
@@ -0,0 +1,166 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import { Constants } from './constants';
+import * as utilities from './utilities';
+
+export class AzureHelpers {
+    constructor() {
+    }
+
+    private credentials = utilities.parseCredentialsFile(null);
+    private token: string = null;
+
+    public obtainAuthorizationToken(cb): void {
+
+        let azure_tenant = this.credentials['AZURE_TENANT'];
+        let azure_client_id = this.credentials['AZURE_CLIENT_ID'];
+        let azure_secret = this.credentials['AZURE_SECRET'];
+        
+        if (this.token != null) {
+            cb(true);
+            return;
+        }
+
+        // XXX - check if credentials are valid
+
+        //url: https://login.windows.net/{{ lookup('env','AZURE_TENANT') }}/oauth2/token
+        //method: POST
+        //body: resource=https%3A%2F%2Fmanagement.core.windows.net%2F&client_id={{ lookup('env','AZURE_CLIENT_ID') }}&grant_type=client_credentials&client_secret={{ lookup('env','AZURE_SECRET') }}
+        //return_content: yes
+        //headers:
+        //  Content-Type: application/x-www-form-urlencoded        
+        var https = require('https');
+        let __this = this;
+
+        let postData: string =  "resource=https%3A%2F%2Fmanagement.core.windows.net%2F&client_id=" + azure_client_id + "&grant_type=client_credentials&client_secret=" + azure_secret;
```

Comment:
are you sure to post user's secret as plaintext over internet? it's better to do some encryption

Reply:
it's better to use some azure sdk lib to do auth?